opm/autodiff/WellInterface_impl.hpp

```diff
@@ -1179,16 +1182,37 @@ namespace Opm
         bool converged;
         WellState well_state0 = well_state;
         do {
-            assembleWellEq(ebosSimulator, dt, well_state, deferred_logger);
+            int exception_thrown = 0;
+            try {
+                assembleWellEq(ebosSimulator, dt, well_state, deferred_logger);
+            } catch (...) {
+                exception_thrown = 1;
+            }
+
+            OPM_CHECK_FOR_EXCEPTIONS_AND_THROW(exception_thrown, "assembleWellEq in WellInterface::solveWellEqUntilConverged() failed.");
```

Comment:
It is always wrong to communicate within the "single well" classes (WellInterface and its derived classes). Errors here must be caught in the BlackoilWellModel class.

Reply:
you are right.